### ISSUE TITLE
cgroups error with Fedora 32 (cgroups v2)

### ISSUE THREAD

**Issue by underhood** [netdata/netdata#8907]
##### Bug report summary
When running default install on Fedora 32 VM the cgroups errors are in the logs and `systemd services` metric group is gone (compared to Fed 31):
```
2020-05-05 14:12:28: netdata INFO  : PLUGIN[cgroups] : thread created with task id 16901
2020-05-05 14:12:28: netdata INFO  : PLUGIN[cgroups] : set name of thread 16901 to PLUGIN[cgroups]
2020-05-05 14:12:28: netdata ERROR : PLUGIN[cgroups] : CGROUP: cannot find cpuacct mountinfo. Assuming default: /sys/fs/cgroup/cpuacct
2020-05-05 14:12:28: netdata ERROR : PLUGIN[cgroups] : CGROUP: cannot find cpuset mountinfo. Assuming default: /sys/fs/cgroup/cpuset
2020-05-05 14:12:28: netdata ERROR : PLUGIN[cgroups] : CGROUP: cannot find blkio mountinfo. Assuming default: /sys/fs/cgroup/blkio
2020-05-05 14:12:28: netdata ERROR : PLUGIN[cgroups] : CGROUP: cannot find memory mountinfo. Assuming default: /sys/fs/cgroup/memory
2020-05-05 14:12:28: netdata ERROR : PLUGIN[cgroups] : CGROUP: cannot find devices mountinfo. Assuming default: /sys/fs/cgroup/devices
2020-05-05 14:12:40: netdata ERROR : PLUGIN[cgroups] : CGROUP: cannot read directory '/sys/fs/cgroup/cpuacct' (errno 2, No such file or directory)
2020-05-05 14:12:40: netdata ERROR : PLUGIN[cgroups] : CGROUP: disabled cpu statistics.
2020-05-05 14:12:40: netdata ERROR : PLUGIN[cgroups] : CGROUP: cannot read directory '/sys/fs/cgroup/blkio' (errno 2, No such file or directory)
2020-05-05 14:12:40: netdata ERROR : PLUGIN[cgroups] : CGROUP: disabled blkio statistics.
2020-05-05 14:12:40: netdata ERROR : PLUGIN[cgroups] : CGROUP: cannot read directory '/sys/fs/cgroup/memory' (errno 2, No such file or directory)
2020-05-05 14:12:40: netdata ERROR : PLUGIN[cgroups] : CGROUP: disabled memory statistics.
2020-05-05 14:12:40: netdata ERROR : PLUGIN[cgroups] : CGROUP: cannot read directory '/sys/fs/cgroup/devices' (errno 2, No such file or directory)
2020-05-05 14:12:40: netdata ERROR : PLUGIN[cgroups] : CGROUP: disabled devices statistics.
2020-05-05 14:13:32: netdata INFO  : MAIN : EXIT: Stopping master thread: PLUGIN[cgroups]
2020-05-05 14:13:32: netdata INFO  : PLUGIN[cgroups] : cleaning up...
2020-05-05 14:13:32: netdata INFO  : PLUGIN[cgroups] : thread with task id 16901 finished
2020-05-05 14:13:53: netdata INFO  : PLUGIN[cgroups] : thread created with task id 1098
2020-05-05 14:13:53: netdata INFO  : PLUGIN[cgroups] : set name of thread 1098 to PLUGIN[cgroups]
2020-05-05 14:13:53: netdata ERROR : PLUGIN[cgroups] : CGROUP: cannot find cpuacct mountinfo. Assuming default: /sys/fs/cgroup/cpuacct
2020-05-05 14:13:53: netdata ERROR : PLUGIN[cgroups] : CGROUP: cannot find cpuset mountinfo. Assuming default: /sys/fs/cgroup/cpuset
2020-05-05 14:13:53: netdata ERROR : PLUGIN[cgroups] : CGROUP: cannot find blkio mountinfo. Assuming default: /sys/fs/cgroup/blkio
2020-05-05 14:13:53: netdata ERROR : PLUGIN[cgroups] : CGROUP: cannot find memory mountinfo. Assuming default: /sys/fs/cgroup/memory
2020-05-05 14:13:53: netdata ERROR : PLUGIN[cgroups] : CGROUP: cannot find devices mountinfo. Assuming default: /sys/fs/cgroup/devices
2020-05-05 14:15:22: netdata INFO  : MAIN : EXIT: Stopping master thread: PLUGIN[cgroups]
2020-05-05 14:15:22: netdata INFO  : PLUGIN[cgroups] : cleaning up...
2020-05-05 14:15:22: netdata INFO  : PLUGIN[cgroups] : thread with task id 1098 finished
2020-05-05 14:15:23: netdata INFO  : PLUGIN[cgroups] : thread created with task id 2948
2020-05-05 14:15:23: netdata INFO  : PLUGIN[cgroups] : set name of thread 2948 to PLUGIN[cgroups]
2020-05-05 14:15:23: netdata ERROR : PLUGIN[cgroups] : CGROUP: cannot find cpuacct mountinfo. Assuming default: /sys/fs/cgroup/cpuacct
2020-05-05 14:15:23: netdata ERROR : PLUGIN[cgroups] : CGROUP: cannot find cpuset mountinfo. Assuming default: /sys/fs/cgroup/cpuset
2020-05-05 14:15:23: netdata ERROR : PLUGIN[cgroups] : CGROUP: cannot find blkio mountinfo. Assuming default: /sys/fs/cgroup/blkio
2020-05-05 14:15:23: netdata ERROR : PLUGIN[cgroups] : CGROUP: cannot find memory mountinfo. Assuming default: /sys/fs/cgroup/memory
2020-05-05 14:15:23: netdata ERROR : PLUGIN[cgroups] : CGROUP: cannot find devices mountinfo. Assuming default: /sys/fs/cgroup/devices
2020-05-05 14:15:34: netdata ERROR : PLUGIN[cgroups] : CGROUP: cannot read directory '/sys/fs/cgroup/cpuacct' (errno 2, No such file or directory)
2020-05-05 14:15:34: netdata ERROR : PLUGIN[cgroups] : CGROUP: disabled cpu statistics.
2020-05-05 14:15:34: netdata ERROR : PLUGIN[cgroups] : CGROUP: cannot read directory '/sys/fs/cgroup/blkio' (errno 2, No such file or directory)
2020-05-05 14:15:34: netdata ERROR : PLUGIN[cgroups] : CGROUP: disabled blkio statistics.
2020-05-05 14:15:34: netdata ERROR : PLUGIN[cgroups] : CGROUP: cannot read directory '/sys/fs/cgroup/memory' (errno 2, No such file or directory)
2020-05-05 14:15:34: netdata ERROR : PLUGIN[cgroups] : CGROUP: disabled memory statistics.
2020-05-05 14:15:34: netdata ERROR : PLUGIN[cgroups] : CGROUP: cannot read directory '/sys/fs/cgroup/devices' (errno 2, No such file or directory)
2020-05-05 14:15:34: netdata ERROR : PLUGIN[cgroups] : CGROUP: disabled devices statistics.
2020-05-05 14:19:30: netdata INFO  : MAIN : EXIT: Stopping master thread: PLUGIN[cgroups]
2020-05-05 14:19:30: netdata INFO  : PLUGIN[cgroups] : cleaning up...
2020-05-05 14:19:30: netdata INFO  : PLUGIN[cgroups] : thread with task id 2948 finished
2020-05-05 14:19:36: netdata INFO  : PLUGIN[cgroups] : thread created with task id 13889
2020-05-05 14:19:36: netdata INFO  : PLUGIN[cgroups] : set name of thread 13889 to PLUGIN[cgroups]
2020-05-05 14:19:36: netdata ERROR : PLUGIN[cgroups] : CGROUP: cannot find cpuacct mountinfo. Assuming default: /sys/fs/cgroup/cpuacct
2020-05-05 14:19:36: netdata ERROR : PLUGIN[cgroups] : CGROUP: cannot find cpuset mountinfo. Assuming default: /sys/fs/cgroup/cpuset
2020-05-05 14:19:36: netdata ERROR : PLUGIN[cgroups] : CGROUP: cannot find blkio mountinfo. Assuming default: /sys/fs/cgroup/blkio
2020-05-05 14:19:36: netdata ERROR : PLUGIN[cgroups] : CGROUP: cannot find memory mountinfo. Assuming default: /sys/fs/cgroup/memory
2020-05-05 14:19:36: netdata ERROR : PLUGIN[cgroups] : CGROUP: cannot find devices mountinfo. Assuming default: /sys/fs/cgroup/devices
2020-05-05 14:19:47: netdata ERROR : PLUGIN[cgroups] : CGROUP: cannot read directory '/sys/fs/cgroup/cpuacct' (errno 2, No such file or directory)
2020-05-05 14:19:47: netdata ERROR : PLUGIN[cgroups] : CGROUP: disabled cpu statistics.
2020-05-05 14:19:47: netdata ERROR : PLUGIN[cgroups] : CGROUP: cannot read directory '/sys/fs/cgroup/blkio' (errno 2, No such file or directory)
2020-05-05 14:19:47: netdata ERROR : PLUGIN[cgroups] : CGROUP: disabled blkio statistics.
2020-05-05 14:19:47: netdata ERROR : PLUGIN[cgroups] : CGROUP: cannot read directory '/sys/fs/cgroup/memory' (errno 2, No such file or directory)
2020-05-05 14:19:47: netdata ERROR : PLUGIN[cgroups] : CGROUP: disabled memory statistics.
```

##### OS / Environment
Fedora 32 AMD64

##### Netdata version
`go-live` branch

##### Component Name
agent/PLUGIN[cgroups]

##### Steps To Reproduce
Install default netdata as root on Fedora 32.

##### Expected behavior
It works.

**Comment by thanasisk** [comment]
Hello @underhood - how did you install netdata? I am trying to reproduce on my FC32 box.

**Comment by underhood** [comment]
Fresh VM Fedora workstation 32 64bit. Install all the build dependencies `dnf install zlib-devel libuuid-devel libuv-devel lz4-devel Judy-devel openssl-devel libmnl-devel gcc make git autoconf autoconf-archive autogen automake pkgconfig curl findutils python cmake`.  Clone from git and then install with `netdata-installer.sh --disable-telemetry`

**Comment by thanasisk** [comment]
Starting from FC31, IIRC, Fedora is using cgroups v2 by default. This breaks a lot of software by default.

A temporary workaround is to instruct the kernel to fall-back to cgroups v1 - you can do this by the following commands:

`sudo dnf install grubby`

`sudo grubby --update-kernel=ALL --args="systemd.unified_cgroup_hierarchy=0"`

Doing so, took care of cgroups related errors in my FC32 (VM) installation.

**Comment by thanasisk** [comment]
Hi there @underhood 
I did a bit of research. Fresh FC32 install, netdata installed as per your instructions, I was able to reproduce the error. It appears to be `cgroups v2` related.

Once I have setup my FC32 installation to use `cgroups v1`, the relevant functionality was there and no error messages in the logs.

What puzzles me is that FC31, which according to what you say, runs normally - to the best of my knowledge v2 is enabled by default in FC31.

**Comment by underhood** [comment]
@thanasisk ok so most probably it doesn't work on Ferora 31 as well out of the box. I force cgroups v1 in my Fed31 system *(which is my main devel workstation) due to docker (and I guess lot of other people do from team here)*

Thanks for helping figure out the problem

**Comment by skrzyp1** [comment]
Have you tried using: https://github.com/netdata/netdata/tree/master/collectors/cgroups.plugin#unified-cgroups-cgroups-v2-support
As i remember there was no automatic cgroup type detection when i was working on that - it had to be switched manually.

**Comment by underhood** [comment]
Closed by #9249
